File: junit_5.6.2/src/test/java/com/rocketnine/junitexamples/LeapAgeTests.java
package com.rocketnine.junitexamples;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.time.LocalDate;

class LeapAgeTests {

    @Test
    void someone_born_after_most_recent_leap_year_has_leap_age_0() {
        // Arrange
        LocalDate birthDate = LocalDate.parse("2025-07-04");
        LocalDate currentDate = LocalDate.parse("2025-09-01");

        // Act
        int actual = Hiker.calculateLeapAge(birthDate, currentDate);

        // Assert
        Assertions.assertEquals(0, actual);
    }

    @Test
    void someone_born_after_most_recent_leap_year_has_leap_age_0_CONDENSED() {
        // Arrange - Act - Assert all in one line
        Assertions.assertEquals(0, Hiker.calculateLeapAge(LocalDate.parse("2025-07-04"), LocalDate.parse("2025-09-01")));
    }

}

File: nunit_3.12/NUnit3_Tests/LeapAgeTests.cs
using System;
using NUnit.Framework;

namespace NUnit3_Tests
{
    [TestFixture]
    public class LeapAgeTests
    {
        [Test]
        public void Someone_born_after_most_recent_leap_year_has_leap_age_0()
        {
            // Arrange
            DateTime birthDate   = DateTime.Parse("2025-07-04");
            DateTime currentDate =  DateTime.Parse("2025-09-16");

            // Act 
            var actual = Hiker.CalculateLeapAge(birthDate, currentDate);
            
            // Assert
            Assert.That(actual, Is.EqualTo(0));
        } 
    }

    public static class Hiker
    {
        public static int CalculateLeapAge(DateTime birthDate, DateTime currentDate)
        {
            return 0;
        }
    }
}